xtask/src/ci/target_matrix.rs

```diff
@@ -37,6 +37,7 @@ pub(crate) fn run(message: String, author: String) -> Result<(), color_eyre::Rep
 }
 
 #[derive(Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
```

Comment:
deny here is kinda pointless

Reply:
I think this is fine: I just tried it with `--build-std` and it was properly denied.

Reply:
that doesn't come from here though, it comes from the deny on `CiTarget`

`deny_unknown_fields` is only applied on deserialization

Reply:
I don't really see an issue then, because it's better to fail fast in general, so even if this isn't needed, it might be useful after refactoring.